### PR TITLE
Allow pagination in WKP Popup, when several features are clicked

### PR DESCRIPTION
**Pull request by edellucien**
Require this react-spatial MR to be merged: https://github.com/geops/react-spatial/pull/223

### CODE REVIEW

File: src/components/Popup/Popup.js [hunk 1]
```diff
@@ -21,46 +25,114 @@ const defaultProps = {
...
+    };
+
+    if (clickedFeatureInfo) {
+      const { features, layer, coordinate } = clickedFeatureInfo;
+      this.componentName = popupComponents[layer.getKey()];
+      this.popupCoordinate =
+        features.length && features[0].getGeometry() instanceof Point
+          ? features[0].getGeometry().getCoordinates()
+          : coordinate;
+    }
```
Comment (oterral) [inline src/components/Popup/Popup.js]: It smells like duplicated code

**Comment by oterral** [comment]
@edellucien the example https://deploy-preview-69--trafimage-maps.netlify.com/#section-bahnhofplanpopup   doesn't work, can you fix it ?

**Comment by edellucien** [comment]
react-spatial version is updated. Example works now :)

File: src/components/Popup/Popup.js [hunk 2]
```diff
@@ -21,46 +25,114 @@ const defaultProps = {
...
-  if (!clickedFeatureInfo) {
-    return null;
+class Popup extends Component {
+  constructor(props) {
+    super(props);
+    const { clickedFeatureInfo } = this.props;
+
+    this.state = {
+      featIndex: 0,
+      featuresList: clickedFeatureInfo ? clickedFeatureInfo.features : [],
```
Comment (mhaertwig) [inline src/components/Popup/Popup.js]: Why do we need this state variable? We already have `clickedFeatureInfo.feature` which updates when necessary.

File: src/components/Popup/Popup.js [hunk 3]
```diff
@@ -21,46 +25,114 @@ const defaultProps = {
...
+      this.componentName = null;
+      this.popupCoordinate = null;
+      this.setState({
+        featIndex: 0,
+        featuresList: [],
+      });
+    }
+  }
+
+  renderPagination(feats) {
```
Comment (mhaertwig) [inline src/components/Popup/Popup.js]: please do not invent multiple names for the same variable.

File: src/components/Popup/Popup.js [hunk 4]
```diff
@@ -21,46 +25,114 @@ const defaultProps = {
...
+
+    const { features, layer, coordinate } = clickedFeature;
+    this.componentName = popupComponents[layer.getKey()];
+    this.popupCoordinate =
+      features.length && features[featIndex].getGeometry() instanceof Point
+        ? features[featIndex].getGeometry().getCoordinates()
+        : coordinate;
+  }
+
+  setFeaturesList(clickedFeature) {
```
Comment (oterral) [inline src/components/Popup/Popup.js]: same here an updateState name seems more approriate

File: src/components/Popup/Popup.scss [hunk 5]
```diff
@@ -1,4 +1,37 @@
-// placeholder
-.wkp-popup {
-  display: block;
+.tm-popup {
+  div:first-child {
+    -webkit-animation: fadein 0.5s; /* Safari, Chrome and Opera > 12.1 */
+    -moz-animation: fadein 0.5s; /* Firefox < 16 */
+    -ms-animation: fadein 0.5s; /* Internet Explorer */
+    -o-animation: fadein 0.5s; /* Opera < 12.1 */
```
Comment (oterral) [inline src/components/Popup/Popup.scss]: Autoprefixer takes care of adding this for you so no need to write css with prefix.

File: src/components/Popup/Popup.scss [hunk 6]
```diff
@@ -1,4 +1,37 @@
...
+      height: 25px;
+      width: 25px;
+
+      .wkp-popup-pagination-button {
+        height: 25px;
+        width: 25px;
+
+        svg {
+          height: 25px;
+          width: 25px;
```
Comment (oterral) [inline src/components/Popup/Popup.scss]: if paranet and parent of parent have the same width , I guess setting width: 100% is more useful.

**Comment by oterral** [comment]
@mhaertwig  can you re review please ?

**Pull request merged**